### PR TITLE
Support `disable-inline-width` attribute when applying layout sizes

### PR DESCRIPTION
**Pull request by caroqliu**
This allows specifying the `sizes` on elements without getting the sizing behavior of sizes in AMP (which adds a width based on which media query matches). Addresses #21736

See [discussion doc](https://docs.google.com/document/d/1qRF38BwZPfgIdNfZ8W1sf1lHyWYd2Y5ZFM4nPSMFt3Y/edit#) for more

### CODE REVIEW

File: spec/amp-html-layout.md [hunk 1]
```diff
@@ -160,6 +160,26 @@ In the following example, if the viewport is wider than `320px`, the image will
 </amp-img>
 ```
 
+### `disable-inline-width`
+
+All AMP elements that support the `responsive` layout, also support the `disable-inline-width` attribute. The presence of this attribute describes how the `sizes` attribute is used. When pairing `disable-inline-width` with `sizes`, the AMP element will propagate the value of `sizes` to the element's underlying tag, as with the `img` nested inside an `amp-img`, **without** getting the sizing behavior of `sizes` in AMP (which adds a width based on which media query matches).
```
Comment (jridgewell) [inline spec/amp-html-layout.md]: I think this needs to start by saying `sizes` (without `disable-inline-width`) will set an inline `width` style on the element.

File: src/custom-element.js [hunk 2]
```diff
@@ -694,6 +694,10 @@ function createBaseCustomElementClass(win) {
         );
       }
 
+      if (this.hasAttribute('disable-inline-width')) {
```
Comment (jridgewell) [inline src/custom-element.js]: This blocks `heights` behavior. Additionally, we'd need to set `sizeList_ = null`. I think this should just be handled as part of the ternary in the sizes block below.

**Comment by westonruter** [comment]
Is this related to #17053? does the presence of `disable-inline-width` allow for the browser's underlying `sizes` behavior to be retained?

**Comment by caroqliu** [comment]
@westonruter Yes, this is related as it directly resolves the original issue filed:

> This might be the expected behavior. But it looks like the <amp-img> sizes attribute causes inline styling to be added to the <amp-img>, which can distort it.
> 
> And this seems to be different from the non-AMP behavior of a sizes attribute in an .

**Comment by westonruter** [comment]
@caroqliu I think I may have identified something that was missed with support for `disable-inline-width` and that has to do with this CSS rule in `ampshared.css`:

https://github.com/ampproject/amphtml/blob/97099ce8bf4826a08d964e7a02da55a7641b1090/css/ampshared.css#L59-L65

Should not that last selector be changed to exclude elements that have the `disable-inline-width` attribute? Like so:

```diff
- [width][height][sizes]:not(.i-amphtml-layout-responsive)
+ [width][height][sizes]:not([disable-inline-width]):not(.i-amphtml-layout-responsive)
```

Otherwise, a regression is introduced by adopting `disable-inline-width` instead of removing the `sizes` attribute: https://github.com/ampproject/amp-wp/pull/4622#issuecomment-618632173.

**Pull request merged**